src/nupic/algorithms/Connections.hpp

```diff
@@ -51,9 +51,9 @@ namespace nupic
       typedef Real32 Permanence;
       typedef UInt64 Iteration;
 
-      #define CELL_MAX (USHRT_MAX-1)
-      #define SEGMENT_MAX (UCHAR_MAX-1)
-      #define SYNAPSE_MAX (UCHAR_MAX-1)
+      // Defaults
+      static const unsigned char MAX_SEGMENTS_PER_CELL = 255;
```

Comment:
This should be `UInt` or something other than `unsigned char`. Otherwise we couldn't change it to be larged than 255. But also semantically this is an unsigned integer.